src/payment/strategies/cybersource/cardinal-client.ts

```diff
@@ -0,0 +1,190 @@
+import { includes } from 'lodash';
+
+import Address from '../../../address/address';
+import {
+    MissingDataError, MissingDataErrorType, NotInitializedError,
+    NotInitializedErrorType, StandardError
+} from '../../../common/error/errors';
+import { CreditCardInstrument, ThreeDSecureToken } from '../../payment';
+import { ThreeDsResult } from '../../payment-response-body';
+
+import {
+    CardinalAccount,
+    CardinalAddress,
+    CardinalConsumer,
+    CardinalEventType,
+    CardinalInitializationType,
+    CardinalOrderData,
+    CardinalPartialOrder,
+    CardinalPaymentBrand,
+    CardinalScriptLoader,
+    CardinalSignatureValidationErrors,
+    CardinalSDK,
+    CardinalTriggerEvents,
+    CardinalValidatedAction,
+    CardinalValidatedData
+} from './index';
+
+export default class CardinalClient {
+    private _sdk?: Promise<CardinalSDK>;
+
+    constructor(
+        private _scriptLoader: CardinalScriptLoader
+    ) {}
+
+    initialize(testMode: boolean): Promise<void> {
+        if (!this._sdk) {
+            this._sdk = this._scriptLoader.load(testMode);
+        }
+
+        return this._sdk.then(() => {});
+    }
+
+    configure(clientToken: string): Promise<void> {
+        return this._getClientSDK()
+            .then(client => new Promise<void>((resolve, reject) => {
+                client.on(CardinalEventType.SetupCompleted, () => {
+                    client.off(CardinalEventType.SetupCompleted);
+                    client.off(CardinalEventType.Validated);
+
+                    resolve();
+                });
+
+                client.on(CardinalEventType.Validated, (data: CardinalValidatedData) => {
+                    client.off(CardinalEventType.SetupCompleted);
+                    client.off(CardinalEventType.Validated);
+
+                    switch (data.ActionCode) {
```

Comment:
Actually at this point we're registering to the events in order to wait only for the `client.setup` response and according to the Cardinal documentation it will be `CardinalValidatedAction.Error` when there is an error and then [in this code](https://github.com/bigcommerce/checkout-sdk-js/pull/565/files/3fbb1c5fd01517dc43b804fe07e873ea65d19c47#diff-c665773cd21b5694abc297d6000a2a5fR92) we registered again to wait for the other cases that can be reached by `cardinal.continue`

Reply:
Sure 👍